modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -2064,12 +2064,61 @@ public int compare(CodegenProperty one, CodegenProperty another) {
         return m;
     }
 
+    /**
+     * Process oneOf and anyOf models in a composed schema and adds them into
+     * their parent's discriminator map if the oneOf and anyOf models contain
+     * the required discriminator. If they don't contain the required
+     * discriminator or the discriminator is the wrong type then an error is
+     * thrown
+     * @param discriminator The CodegenDiscriminator that we will add models to
+     * @param c The ComposedSchema that contains the discriminator and oneOf/anyOf schemas
+     * @param discPropName The String that is the discriminator propertyName in the schema
+     */
+    private void addComposedMappedModels(CodegenDiscriminator discriminator, ComposedSchema c, String discPropName) {
+        ArrayList<List<Schema>> listOLists = new ArrayList<List<Schema>>();
+        listOLists.add(c.getOneOf());
+        listOLists.add(c.getAnyOf());
+        for (List<Schema> schemaList: listOLists) {
+            if (schemaList == null) {
+                continue;
+            }
+            for  (Schema sc: schemaList) {
+                String modelName = ModelUtils.getSimpleRef(sc.get$ref());
```

Comment:
This will fail, as these schemas may also be defined inline rather than as a $ref.

Reply:
Should I use ModelUtils.getReferencedSchema here?

Reply:
Changed to using ModelUtils.getReferencedSchema.
@jimschubert because inlineModelResolver is invoked, won't inline models always have $refs to the extracted inline schema? If so, then it looks like using `String modelName = ModelUtils.getSimpleRef(sc.get$ref());` would always work here.

Reply:
I'm not sure what you're referring to, but you won't always have a $ref here.

If you're referring to flatten in InlineModelResolver, I don't think that's doing what you think it's doing. Please review https://github.com/OpenAPITools/openapi-generator/blob/ea55968737fa80a3e306d1b08032ff4d911774c9/modules/openapi-generator/src/main/java/org/openapitools/codegen/InlineModelResolver.java#L330

Reply:
To explain further, in case it's not clear... You'll input have a `$ref` if you have a Schema defined (like `#/components/Schemas/Dog`. It's perfectly valid to have a spec with _only_ inline models, and also to have something like allOf, anyOf, or oneOf which define inline models for those compositions. Although there not common, they'll fail here.

Reply:
Hmm, I am having trouble understanding your example.
[Inline OneOf](https://github.com/spacether/openapi-generator/blob/issue_4904_fix_composed_schema_discriminator_map/modules/openapi-generator/src/test/resources/3_0/oneoOfDiscriminator.yaml#L25): we have a schema which has a oneOf schema with an inline definition.
- Are you saying that when defining oneOf schemas I need to use a mix of inline and $ref to hit this null $ref use case?
- Can you provide a full example schema which has a oneOf definition in it where the oneOf schema will have a null $ref value?

Reply:
Are you saying that we can have a model with two levels of inline oneOf definition?
If so we are guaranteed that the first level is extracted to its own model per my explanation above. I will add that as a test case.

Reply:
@jimschubert we now have a test with inline oneOf schemas with inline oneOf schemas which results in null refs. We now continue through and skip any schemas with no ref because it is impossible to build a discriminator map with them.
For now I am resolving this convo because I think that addresses your concern.
If it doesn't please let me know and I'll be happy to make more changes.

Reply:
Yes, I think that addresses the concern. Sorry, I saw your question about providing an example, but I couldn't find the question when I came back to it (probably because of your additional commit).